magicctapipe/scripts/lst1_magic/semi_automatic_scripts/database_production/update_MAGIC_database.py

```diff
@@ -0,0 +1,351 @@
+        new_rows = []
+
+        for index, row in non_matching_rows_reset.iterrows():
+            date = row["DATE"]
+            source = row["source"]
+            run_id = row["Run ID"]
+            run_id = str(run_id)
+
+            date_obj = datetime.strptime(date, "%Y%m%d")
+            date_obj += timedelta(days=1)
+            new_date = datetime.strftime(date_obj, "%Y%m%d")
+            YYYY = new_date[:4]
+            MM = new_date[4:6]
+            DD = new_date[6:8]
+            Y = "_Y_"
+
+            path = f"/fefs/onsite/common/MAGIC/data/M{tel_id}/event/Calibrated/{YYYY}/{MM}/{DD}"
+
+            if os.path.exists(path):
+                files = os.listdir(path)
+                count_with_run_id = 0
+                # Counter for files that include the run_id.
+                for filename in files:
+                    if Y in filename:
+                        if new_date in filename:
+                            if source in filename:
+                                if run_id in filename:
+                                    count_with_run_id += 1
+                if count_with_run_id != 0:
+                    new_rows.append(
+                        {
+                            "DATE": date,
+                            "source": source,
+                            "Run ID": run_id,
+                            "number of subruns": count_with_run_id,
+                        }
+                    )
+
+        new_rows = pd.DataFrame(new_rows)
+        new_rows["DATE"] = pd.to_datetime(new_rows["DATE"])
+        combined_df = pd.concat([DF, new_rows], ignore_index=True)
+        combined_df["DATE"] = pd.to_datetime(combined_df["DATE"], errors="coerce")
+        combined_df = combined_df.sort_values("DATE")
+
+        combined_df["DATE"] = combined_df["DATE"].dt.strftime("%Y%m%d")
+        combined_df["Run ID"] = combined_df["Run ID"].astype(int)
+        combined_df.reset_index(drop=True, inplace=True)
+
+        for column in combined_df.columns[4:]:
+            not_null_data = combined_df[column].dropna()
+            if not_null_data.empty:
+                continue  # Skip if all values are NaN
+
+            inferred_type = pd.api.types.infer_dtype(not_null_data, skipna=True)
+
+            if inferred_type == "boolean":
+                default_value = False
+            elif inferred_type == "integer":
+                default_value = 0
+            elif inferred_type == "floating":
+                default_value = 0.0
+            elif inferred_type == "string":
+                default_value = "NaN"
+            else:
+                continue
+
+            combined_df[column] = (
+                combined_df[column]
+                .fillna(default_value)
+                .astype(type(not_null_data.iloc[0]))
+            )
+
+        combined_df = combined_df.infer_objects()
+
+    return combined_df
+
+def create_new_database(df, date_min, date_max, tel_id):
```

Comment:
this function has a lot of common code with the previous one
This is why I was suggesting to have only one function that has previous table by default as Null, and if so it first creates a new empty table, and then runs updating function on this empty table

Reply:
I see... And I think I just did it :) New version of the script will appear soon.